hooks/lint-guard.ts

```
@@ -0,0 +1,36 @@
+
+function isHookInput(value: unknown): value is HookInput {
+	if (typeof value !== "object" || value === null || !("tool_input" in value)) {
+		return false;
+	}
+
+	// eslint-disable-next-line flawless/naming-convention -- Upstream-defined input shape
+	const { tool_input } = value as { tool_input: unknown };
+	return typeof tool_input === "object" && tool_input !== null && "file_path" in tool_input;
+}
+
+const input: unknown = JSON.parse(readFileSync(0, "utf-8"));
```

ESLint

[error] Unsafe argument of type error typed assigned to a parameter of type `string`.

(ts/no-unsafe-argument)